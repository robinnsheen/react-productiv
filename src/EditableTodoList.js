import React from "react";
import EditableTodo from "./EditableTodo";

/** Show list of editable todos.
 *
 * Props:
 * - todos: array of [ todo, ... ]
 * - update(): fn to call to update a todo
 * - remove(): fn to call to remove a todo
 *
 * TodoApp -> EditableTodoList -> [ EditableTodo, ... ]
 */

function EditableTodoList({todos, update, remove}) {
  console.log("editablelist, todos", todos);
  return (
      <div>
        { /** fix me */}

        <div>{todos.map(todo => <EditableTodo todo={todo} update={update} remove={remove}/>)}</div>
      </div>
  );
}

export default EditableTodoList;
