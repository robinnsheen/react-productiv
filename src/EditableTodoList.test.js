import React from "react";
import { render } from "@testing-library/react";
import EditableTodoList from "./EditableTodoList";

let testTodos = [
  {
    id: 1,
    title: "testTitle1",
    priority: 1,
    description: "test description"
  },
  { id: 2,
    title: "testTitle2",
    priority: 2,
    description: "test description"
  },
  { id: 3,
    title: "testTitle3",
    priority: 3,
    description: "test description"
  }
];

const mockUpdate = jest.fn();
const mockRemove = jest.fn();

describe("EditableTodosList", function () {
  it("renders without crashing", function () {
    render(<EditableTodoList todos={testTodos} update={mockUpdate} remove={mockRemove} />);
  });

  it("matches snapshot", function () {
    const { container } = render(
      <EditableTodoList
        todos={testTodos}
        update={mockUpdate}
        remove={mockRemove} />);
    expect(container).toMatchSnapshot();
  });

  it("renders all editable todos in todos", function() {
    const result = render(
      <EditableTodoList
        todos={testTodos}
        update={mockUpdate}
        remove={mockRemove} />);

    result.debug(result.container);

    expect(result.getByText("testTitle1")).toBeInTheDocument();
    expect(result.getByText("testTitle2")).toBeInTheDocument();
    expect(result.getByText("testTitle3")).toBeInTheDocument();

  })


});
