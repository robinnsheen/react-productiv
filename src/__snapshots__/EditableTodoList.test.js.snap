// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditableTodosList matches snapshot 1`] = `
<div>
  <div>
    <div
      class="EditableTodo"
    >
      <div
        class="mb-3"
      >
        <div
          class="float-end text-sm-end"
        >
          <button
            class="EditableTodo-toggle btn-link btn btn-sm"
          >
            Edit
          </button>
          <button
            class="EditableTodo-delBtn btn-link btn btn-sm text-danger"
          >
            Del
          </button>
        </div>
        <div
          class="Todo"
        >
          <div>
            <b>
              testTitle1
            </b>
             
            <small>
              (priority: 
              1
              )
            </small>
          </div>
          <div>
            <small>
              test description
            </small>
          </div>
        </div>
      </div>
    </div>
    <div
      class="EditableTodo"
    >
      <div
        class="mb-3"
      >
        <div
          class="float-end text-sm-end"
        >
          <button
            class="EditableTodo-toggle btn-link btn btn-sm"
          >
            Edit
          </button>
          <button
            class="EditableTodo-delBtn btn-link btn btn-sm text-danger"
          >
            Del
          </button>
        </div>
        <div
          class="Todo"
        >
          <div>
            <b>
              testTitle2
            </b>
             
            <small>
              (priority: 
              2
              )
            </small>
          </div>
          <div>
            <small>
              test description
            </small>
          </div>
        </div>
      </div>
    </div>
    <div
      class="EditableTodo"
    >
      <div
        class="mb-3"
      >
        <div
          class="float-end text-sm-end"
        >
          <button
            class="EditableTodo-toggle btn-link btn btn-sm"
          >
            Edit
          </button>
          <button
            class="EditableTodo-delBtn btn-link btn btn-sm text-danger"
          >
            Del
          </button>
        </div>
        <div
          class="Todo"
        >
          <div>
            <b>
              testTitle3
            </b>
             
            <small>
              (priority: 
              3
              )
            </small>
          </div>
          <div>
            <small>
              test description
            </small>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
